flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/restore/RocksDBIncrementalRestoreOperation.java

```diff
@@ -105,6 +110,10 @@ public class RocksDBIncrementalRestoreOperation<K> implements RocksDBRestoreOper
 
     private boolean isKeySerializerCompatibilityChecked;
 
+    private final TriConsumerWithException<
+                    Collection<IncrementalLocalKeyedStateHandle>, byte[], byte[], Exception>
+            rescalingRestoreFromLocalStateOperation;
+
```

Comment:
This can be replaced by a boolean flag (`useIngestDbRestoreMode`) and a "normal" if and call in `restoreWithRescaling`.
I think it would be more readable.